library/src/main/java/com/schibsted/spain/barista/interaction/BaristaListInteractions.kt

```diff
@@ -58,11 +61,19 @@ object BaristaListInteractions {
     private fun performMagicAction(@IdRes id: Int?, position: Int, recyclerAction: ViewAction, listViewAction: ViewAction) {
         verifyOneSingleMatch(id)
 
+        val recyclerMatcher = findRecyclerMatcher(id)
+        val listViewMatcher = findListViewMatcher(id)
         val spyFailureHandler = SpyFailureHandler()
         try {
-            performOnRecycler(id, recyclerAction, spyFailureHandler)
+            performOnRecycler(recyclerMatcher, recyclerAction, spyFailureHandler)
         } catch (noRecyclerMatching: NoMatchingViewException) {
-            performOnListView(id, position, listViewAction, spyFailureHandler)
+          try {
+            performOnListView(listViewMatcher, position, listViewAction, spyFailureHandler)
+          } catch (listViewError: Throwable) {
```

Comment:
You are not catching the NoMatchingViewException for the listView ""try"" is this intended?

Reply:
Yes it is. The idea behind this magic is: We don't know if you'r using a Recycler or a ListView, so we'll first try to do it on a recycler, and if we don't find it we'll try on a ListView.

And what if it doesn't find a ListView either? That won't happen, because in that case the method will fail before this code, in the `verifyOneSingleMatch()` invocation. 

The implementation is a bit confusing, I know. But at least it's covered by tests!
```java 
  @Test
  public void fail_whenNoViewFound() {
    launchTestActivity(ListsActivity.buildIntent());

    Throwable thrown = catchThrowable(() -> clickListItem(20));

    spyFailureHandlerRule.assertEspressoFailures(1);
    assertThat(thrown).isInstanceOf(BaristaException.class)
        .hasMessage("No ListView or RecyclerView found in the hierarchy");
  }
```

_See @SmasSive? Thanks to the verbose message in the test we know that we're doing it right_ :p

Reply:
Oh I see, I'm not familiar with the architecture and I thought that something like this would happen!